packages/framework/ai-collab/src/explicit-strategy/agentEditReducer.ts

```diff
@@ -104,7 +125,7 @@ export function applyAgentEdit(
 	idGenerator: IdGenerator,
 	definitionMap: ReadonlyMap<string, SimpleNodeSchema>,
 	validator?: (edit: TreeNode) => void,
-): TreeEdit {
+): { edit: TreeEdit; uiDiff: UiDiff } {
```

Comment:
Question that would be good to clarify in the API docs for this function: how is the returned `edit` object different from the input `treeEdit` object? They are the same type, but we seem to be returning something that is derived from the input, with additional information?

Reply:
I know the edit log design to only expose the necessary information. Like it wouldn't expose index of the node. But the `UiDiff` type needs index

Reply:
hmm...seems like only UI needs `editLog` for now. Can we remove `editLog` type here?

Reply:
Josh's question was a different point though. The function takes a TreeEdit as input and also gives one back in its output, so it would be good for it to clarify what is it doing to it or why does it have to return a new one. That was already an unanswered question before this PR, so I'm inclined to not block this change on it.